src/wasm/wasm-binary.cpp

```diff
@@ -5426,6 +5436,27 @@ void WasmBinaryBuilder::visitBrOnExn(BrOnExn* curr) {
   curr->finalize();
 }
 
+void WasmBinaryBuilder::visitCallRef(CallRef* curr) {
+  BYN_TRACE("zz node: CallRef\n");
+  curr->target = popNonVoidExpression();
+  auto type = curr->target->type;
+  if (!type.isRef()) {
+    throwError("Non-ref type for a call_ref");
+  }
+  auto heapType = type.getHeapType();
+  if (!heapType.isSignature()) {
+    std::cerr << type << '\n';
+    throwError("Invalid reference type for a call_ref");
```

Comment:
How about printing these in a single line, like, `Invalid reference type for a call_ref: type`?

Reply:
I don't see an easy way to do that. We should really have an easy way to convert a type into a string...

I agree it's a little odd as it is. Let me just remove the type for now, to match other errors.

Reply:
It seems we have [`Type::toString`](https://github.com/WebAssembly/binaryen/blob/b2d797f1f9f1192b8f4d2664f76a8d0b9278a0ef/src/wasm/wasm-type.cpp#L117)?

Reply:
Thanks! I missed that, will add...